File: exercise/ts/week01-code-analysis/src/storeAccount.ts
import {Movie} from "./movie";

export class StoreAccount {
    public totalSold: number = 0;
    public allSales: MovieSale[];

    constructor() {
        this.allSales = [];
    }

    sell(movie: Movie, to: string) {
        this.totalSold += movie.unitPrice;
        this.allSales.push(new MovieSale(to, movie.title));
    }
}

class MovieSale {
    public clientName: string;
    public movieName: string;

    constructor(clientName: string, movieName: string) {
        this.clientName = clientName;
        this.movieName = movieName;
    }
}

File: exercise/kotlin/week01/src/main/kotlin/movie/StoreAccount.kt
package movie

class StoreAccount {
    var totalSold: Double = 0.0
    var allSales: MutableList<MovieSale> = mutableListOf()

    fun sell(movie: Movie, to: String) {
        totalSold += movie.unitPrice
        allSales.add(MovieSale(to, movie.title))
    }

    class MovieSale(val clientName: String, val movieName: String)
}
